### DISCUSSION TITLE
Magnetic One Azure authentication

### DISCUSSION THREAD

**Discussion by junkgear** [microsoft/autogen#4212] [Q&A] [answered]
Azure OpenAI service

To configure for Azure OpenAI service, set the following environment variables:

CHAT_COMPLETION_PROVIDER='azure'
CHAT_COMPLETION_KWARGS_JSON with the following JSON structure:
{
  "api_version": "2024-02-15-preview",
  "azure_endpoint": "REPLACE_WITH_YOUR_ENDPOINT",
  "model_capabilities": {
    "function_calling": true,
    "json_output": true,
    "vision": true
  },
  "azure_ad_token_provider": "DEFAULT",
  "model": "gpt-4o-2024-05-13"
}

Where to provide api key for magnetic-one setup

Getting this error:

write a python program for fibonaci series, validate and test it make it production ready
DefaultAzureCredential failed to retrieve a token from the included credentials.
Attempted credentials:
        EnvironmentCredential: EnvironmentCredential authentication unavailable. Environment variables are not fully configured.
Visit https://aka.ms/azsdk/python/identity/environmentcredential/troubleshoot to troubleshoot this issue.
        ManagedIdentityCredential: ManagedIdentityCredential authentication unavailable, no response from the IMDS endpoint.
        SharedTokenCacheCredential: SharedTokenCacheCredential authentication unavailable. No accounts were found in the cache.
        AzureCliCredential: Azure CLI not found on path
        AzurePowerShellCredential: PowerShell is not installed
        AzureDeveloperCliCredential: Azure Developer CLI could not be found. Please visit https://aka.ms/azure-dev for installation instructions and then,once installed, authenticate to your Azure account using 'azd auth login'.

**Comment by ekzhu** [comment, marked as answer]
This one uses azure AAD auth. Can you switch to use key-based auth:

```json
{
"api_version": "2024-02-15-preview",
"azure_endpoint": "REPLACE_WITH_YOUR_ENDPOINT",
"model_capabilities": {
"function_calling": true,
"json_output": true,
"vision": true
},
"api_key": "Your API key",
"model": "gpt-4o-2024-05-13"
}
```

**Reply by jiahau3** [reply]
Thanks for providing key-based auth.
I've found out that naming for model has to be revised for Azure as well.
Here is the JSON worked for using Azure openAI
```json
{
"api_version":"2024-06-01",
"azure_endpoint":"REPLACE_WITH_YOUR_ENDPOINT",
"model_capabilities":
{
"function_calling":true,
"json_output":true,
"vision":true
},
"api_key":"Your API key",
"azure_deployment":"Your Deployment Name"
}
```

**Comment by junkgear** [comment]
Please also provide the correct json format for adding additional parameters like temperature, max_tokens and top_p
